pkg/preflight/aws.go

```diff
@@ -84,10 +86,16 @@ func (a *aws) setAWSCredentialEnvVars(cluster *corev1.StorageCluster) error {
 		}
 	}
 	if accessKeyEnv == nil && secretKeyEnv == nil {
+		// No credential provided or credentials are removed
 		logrus.Debugf("no aws credentials provided, will use instance privileges instead")
+		if a.credentialHash != "" {
+			a.credentialHash = ""
+			a.ops = nil
+			resetAWSCredentialEnvVars()
```

Comment:
probably change the function name to remove or unset. The setAWSCredentialEnvVars function actually resets the value.